dotCMS/src/integration-test/java/com/dotmarketing/business/VersionableAPITest.java

```diff
@@ -0,0 +1,331 @@
+		//Create Template
+		Template template=new Template();
+        template.setTitle("TEST TEMPLATE VERSIONABLE");
+        template.setBody("<html><body> I'm mostly empty </body></html>");
+        template=APILocator.getTemplateAPI().saveTemplate(template, host, user, false);
+        
+        //Call Versionable
+        Versionable verAPI = APILocator.getVersionableAPI().findWorkingVersion(template.getIdentifier(), user, false);
+        
+        //Check same Template
+        assertEquals(verAPI.getTitle(),template.getTitle());
+        assertEquals(verAPI.getInode(),template.getInode());
+        
+        //Delete Template
+        APILocator.getTemplateAPI().delete(template, user, false);
+	}
+	
+	@Test
+	public void testFindWorkingVersionFile() throws Exception{
+		//Create File
+		String folderName = "/testOldFile"+UUIDGenerator.generateUuid();
+	    Folder ff = APILocator.getFolderAPI().createFolders(folderName, host, user, false);
+	    
+	    // file data in tmp folder
+	    java.io.File tmp=new java.io.File(APILocator.getFileAPI().getRealAssetPathTmpBinary()+java.io.File.separator+"testOldFile");
+	    if(!tmp.exists()) tmp.mkdirs();
+	    java.io.File data=new java.io.File(tmp,
+	            "test-"+UUIDGenerator.generateUuid()+".txt");
+	    
+	    FileWriter fw=new FileWriter(data,true);
+	    fw.write("file content"); fw.close();
+	    
+	    // legacy file creation
+	    File file = new File();
+	    file.setFileName("legacy.txt");
+	    file.setFriendlyName("legacy.txt");
+	    file.setMimeType("text/plain");
+	    file.setTitle("legacy.txt");
+	    file.setSize((int)data.length());
+	    file.setModUser(user.getUserId());
+	    file.setModDate(Calendar.getInstance().getTime());
+	    file = APILocator.getFileAPI().saveFile(file, data, ff, user, false);
+	    
+	    //Call Versionable
+	    Versionable verAPI = APILocator.getVersionableAPI().findWorkingVersion(file.getIdentifier(), user, false);
+	    
+	    //Check same File
+        assertEquals(verAPI.getTitle(),file.getTitle());
+        assertEquals(verAPI.getInode(),file.getInode());
+        
+        //Delete File
+        APILocator.getFileAPI().delete(file, user, false);
+        APILocator.getFolderAPI().delete(ff, user, false);
+	}
+	
+	@Test(expected = DotDataException.class)
+	public void testFindWorkingVersionContentlet() throws Exception{
+		//Create Contentlet
+		Structure testStructure = APILocator.getStructureAPI().findByVarName("webPageContent", user);
+
+        Contentlet cont=new Contentlet();
+        cont.setStructureInode(testStructure.getInode());
+        cont.setStringProperty("title", "Testing Working");
+        cont.setStringProperty("body", "TESTING");
+        cont.setStringProperty("contentHost", host.getIdentifier());
+        cont.setReviewInterval( "1m" );
+        cont.setStructureInode( testStructure.getInode() );
+        cont.setHost( host.getIdentifier() );
+        cont = APILocator.getContentletAPI().checkin(cont, user, false);
```

Comment:
code from 173 to 181 can be replaced with:
Contentlet contentlet = new ContentletDataGen(testStructure.getInode()).nextPersisted();